src/Microsoft.OData.Edm/Csdl/CsdlReader.Json.cs

```diff
@@ -0,0 +1,201 @@
+        /// <param name="context">The parser context.</param>
+        /// <returns>A list of CsdlModel (no semantics) of the referenced models.</returns>
+        private static List<CsdlModel> LoadReferencedCsdl(CsdlModel csdlModel, JsonParserContext context)
+        {
+            List<CsdlModel> referencedAstModels = new List<CsdlModel>();
+            if (context.Settings.JsonReaderFactory == null)
+            {
+                // don't try to load CSDL-JSON doc, but this.edmReferences's namespace-alias need to be used later.
+                return referencedAstModels;
+            }
+
+            foreach (var edmReference in csdlModel.CurrentModelReferences)
+            {
+                // If nothing included, why does it exist?
+                if (!edmReference.Includes.Any() && !edmReference.IncludeAnnotations.Any())
+                {
+                    continue;
+                }
+
+                // Skip the built-in vocabulary annotation model
+                if (edmReference.Uri != null && (edmReference.Uri.OriginalString.EndsWith("/Org.OData.Core.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Capabilities.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Authorization.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Validation.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/Org.OData.Community.V1.json", StringComparison.Ordinal) ||
+                    edmReference.Uri.OriginalString.EndsWith("/OData.Community.Keys.V1.json", StringComparison.Ordinal)))
+                {
+                    continue;
+                }
+
+                Utf8JsonReader referencedJsonReader = context.Settings.JsonReaderFactory(edmReference.Uri, out bool skip);
+                if (!skip)
+                {
+                    string source = edmReference.Uri != null ? edmReference.Uri.OriginalString : null;
+                    CsdlJsonReaderSettings newSettings = context.Settings.Clone();
+
+                    // set it to null to make sure stop the next level reference parsing.
+                    newSettings.JsonReaderFactory = null;
+                    JsonParserContext subContext = new JsonParserContext(newSettings, source);
+
+                    CsdlModel subCsdlModel = CsdlJsonParser.ParseCsdlDocument(ref referencedJsonReader, subContext);
+                    if (subCsdlModel != null && subContext.IsSuccessed())
+                    {
+                        Version referencedVersion = subCsdlModel.CsdlVersion;
+                        if (!csdlModel.CsdlVersion.Equals(referencedVersion))
+                        {
+                            // Should we report error for this, or just ignore it?
+                            ;

```

Comment:
I think it's fine to ignore. There may be scenarios where referenced schemas are at a different version, and that's okay.

Reply:
Cool, I removed the codes and added the comments for further maintenance. 

---
In reply to: [429396034](https://github.com/OData/odata.net/pull/1775#discussion_r429396034) [](ancestors = 429396034)